### PR TITLE
Add Business services card

### PR DESCRIPTION
**Pull request by raja651**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
This PR adds the new Business card to the in app-marketplace and renders the business card for the places where business card products would be shown. 

Closes https://github.com/Automattic/woocommerce.com/issues/19974.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Switch the WCCOM branch to the one in this PR https://github.com/Automattic/woocommerce.com/pull/20361 and rebuild the assets.
2. run `npm run wp:cli cache flush` on WCCOM branch for the extension API's to have the new fields that is required for business services card
3.  Make sure you have some business services product on your local WCCOM. you can confirm by going to https://woocommerce.test/wp-json/wccom-extensions/1.0/search?category=business-services&country=US and confirm response has some product, otherwise add `Business services` products on your local WCCOM
4. Now go back to Core branch  and In plugins/woocommerce-admin/client/marketplace/components/constants.ts, edit the MARKETPLACE_HOST const to be https://woocommerce.test
5. Go to /plugins/woocommerce and run pnpm --filter='@woocommerce/plugin-woocommerce' build
6. Go to `Business services` [tab](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=business-services&path=%2Fextensions) 
7. confirm the business services card shown according to Figma [design](https://www.figma.com/file/5jyiYtoTF2ZUxgkulPVpOh/Business-Services?node-id=3-3406&mode=dev) for desktop and mobile
8. Test around this Issue
Screenshot 2024-05-06 at 6 23 49 PM

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Review by KokkieH — CHANGES_REQUESTED** [review]
The cards are correctly showing for Business Service products in both search and Discover/no results tabs, but there are some visual and functionality issues with the cards themselves:

It seems bigger (taller) featured images can break out of the card:

Extensions_‹_WooCommerce_‹_woocommerce

Figma has the description section at 150px high, but currently on the cards it's 242px. Total card height in Figma 439px compared to 530px on the cards.

Figma has a 1px wide horizontal line/border added below the featured image.

Lastly, the cards aren't clickable. They should be linking back to the product page on WCCOM (which in turn redirects to the service provider).

**Comment by mcliwanow** [comment]
> It seems bigger (taller) featured images can break out of the card:

I would suggest using image transformation params to ensure we don't load too big images: https://docs.wpvip.com/vip-file-system/image-files/transformations/ (apart from a CSS fix, as we should have a fallback in case image is not properly resized on the server).

**Comment by raja651** [comment]
> Figma has the description section at 150px high, but currently on the cards it's 242px. Total card height in Figma 439px compared to 530px on the cards.
> 
> Figma has a 1px wide horizontal line/border added below the featured image.

Added border and fixed the height issue for image and descriptions. The description height and card is dynamically adjusted to the image size. so fixing taller image issue fixes that as well in [ba04d6a](https://github.com/woocommerce/woocommerce/pull/47183/commits/ba04d6a5452e1a54f5c4e71d4293b752d539d07f) 

> I would suggest using image transformation params to ensure we don't load too big images: https://docs.wpvip.com/vip-file-system/image-files/transformations/ (apart from a CSS fix, as we should have a fallback in case image is not properly resized on the server).

Added useEffect to look the image size and resize it if its larger than 288px that way we don't have enlarge the smaller images and blur it . Done in [ba04d6a](https://github.com/woocommerce/woocommerce/pull/47183/commits/ba04d6a5452e1a54f5c4e71d4293b752d539d07f)

> Lastly, the cards aren't clickable. They should be linking back to the product page on WCCOM (which in turn redirects to the service provider).

I thought this was already done in other tasks, i will add that part of this task before i could do that i also wanted to check if we have to fire any tracks event similar to the one on WCCOM on the card click here?

**Comment by KokkieH** [comment]
> I thought this was already done in other tasks

It was working before, but that's because these were using regular product cards, which this PR replaces with the new Business Services cards :)

**Comment by mcliwanow** [comment]
> wanted to check if we have to fire any tracks event similar to the one on WCCOM on the card click here?

We should fire the same event as we do for other cards:

https://github.com/woocommerce/woocommerce/blob/ba04d6a5452e1a54f5c4e71d4293b752d539d07f/plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.tsx#L217-L232

**Comment by raja651** [comment]
Added the product url to make it clickable and added tracks event to Business card in [db6f13f](https://github.com/woocommerce/woocommerce/pull/47183/commits/db6f13f5e396069cb5033f8f64756789d1e89ced)

File: plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss [hunk 1]
```diff
@@ -284,6 +284,68 @@
...
+					flex: 1 0 0;
+					flex-direction: column;
+					gap: 4px;
+
+					h2 {
+						margin: 0;
+					}
+
+					p {
+						font-size: 14px;
```
Comment (KokkieH) [inline plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss]:
```suggestion
						font-size: 13px;
```
According to Figma

File: plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss [hunk 2]
```diff
@@ -284,6 +284,68 @@
...
+						font-weight: normal;
+					}
+				}
+
+				&__badge {
+					> span {
+						font-weight: 400;
+						font-size: 12px;
```
Comment (KokkieH) [inline plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss]:
```suggestion
						font-size: 11px;
```

According to Figma

File: plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss [hunk 3]
```diff
@@ -284,6 +284,68 @@
...
+
+				&__main-content {
+					display: flex;
+					flex: 1 0 0;
+					flex-direction: column;
+					gap: 4px;
+
+					h2 {
+						margin: 0;
+					}
```
Comment (KokkieH) [inline plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.scss]:
```suggestion
					h2 {
						margin: 0;
						font-size: 16px;
					}
```
According to Figma

**Review by KokkieH — APPROVED** [review]
Looking good. Thanks for the changes, Raja.

Just some small nitpicks about font sizes :)

I also confirmed that the Tracks event is being triggered. For other testers, run `localStorage.setItem( 'debug', 'wc-admin:*' );` in the console, then view localStorage and Cmd-Click a card. An item should be logged there containing what would be sent as a Tracks event on a live site.

**Comment by raja651** [comment]
> Just some small nitpicks about font sizes :)

Fixed font sizes in [b1f36fe](https://github.com/woocommerce/woocommerce/pull/47183/commits/b1f36fe8613e766dc72926d415859df4ab813680)

**Pull request merged**